### PR TITLE
Certification tests for MemCached State Store component.

### PR DESCRIPTION
**Pull request by tmacam**
# Description

Adds certification tests for MemCached State Store component.

* Fixes a problem in memcache state store parsing TTL values.

## Issue reference

Please reference the issue this PR will close: #1929

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by berndverst — CHANGES_REQUESTED** [review]
gofumpt your files

File: state/memcached/memcached.go [hunk 1]
```diff
@@ -125,6 +130,10 @@ func (m *Memcached) parseTTL(req *state.SetRequest) (*int32, error) {
 		}
 		parsedInt := int32(parsedVal)
 
+		if parsedInt < 0 {
```
Comment (berndverst) [inline state/memcached/memcached.go:133]: In general negative TTL means to persist the item in Dapr State Store APIs. Can we change this to not through an error but persist the item instead?

Comment (berndverst) [reply state/memcached/memcached.go:133]: Since it seems `0` means not expiring, just remap `<1` to `0`

Comment (tmacam) [reply state/memcached/memcached.go:133]: Sure. I will replace a -1 with 0 (as that is the value for no expiration on memcached) and error on any other negative number.

Comment (tmacam) [reply state/memcached/memcached.go:133]: I will keep 0 as a pass through. I guess if you are a memcached user you expect 0 to means "persist" and be equivalent to Dapr's -1

File: tests/certification/state/memcached/go.mod [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	sigs.k8s.io/yaml v1.3.0 // indirect
+)
+
+replace github.com/dapr/components-contrib/tests/certification => ../../
+
+replace github.com/dapr/components-contrib => ../../../../
+
+replace github.com/dapr/go-sdk => github.com/hunter007/dapr-go-sdk v1.3.1-0.20220709114046-2f2dc4f9a684
+
+replace github.com/dapr/dapr => github.com/1046102779/dapr v1.5.2-0.20220829014128-56ac94bfadd1
```
Comment (berndverst) [inline tests/certification/state/memcached/go.mod:133]: Remove this

File: tests/certification/state/memcached/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+3. Negative test to fetch record with key, that is not present.
+
+## Test save or update data with different TTL settings:
+1. TTL not expiring ([`0` for memcached](https://github.com/memcached/memcached/wiki/Commands#set))
+2. TTL not a valid number
+3. Provide a TTL of 1 second:
+    1. Fetch this record just after saving
+    2. Sleep for 2 seconds
+    3. Try to fetch again after a gap of 2 seconds, record shouldn't be found
+
```
Comment (berndverst) [inline tests/certification/state/memcached/README.md:23]: Please add a test where you:
1. Set a key to show the connection is fine. Make the TTL 2 minutes or something.
2. Interrupt the network (the memcache ports) for say 60 seconds or something.
3. Wait 10 seconds.
4. Try to read the key you wrote earlier and assert the value.

File: tests/certification/state/memcached/go.mod (hunk 2 above)
Comment (berndverst) [reply tests/certification/state/memcached/go.mod:133]: Just the dapr/dapr replacement to be clear

Comment (tmacam) [reply tests/certification/state/memcached/go.mod]: Done on 8acd78e45ad1d11134d758a628d1030f54b8de75

File: tests/certification/state/memcached/README.md (hunk 3 above)
Comment (tmacam) [reply tests/certification/state/memcached/README.md:23]: fixed in d3a7b524976cad00a6023fb342263dfeafc94093

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
@tmacam please update docs to mark the component stable

**Pull request merged**